src/reflect/scala/reflect/internal/transform/Erasure.scala

```diff
@@ -134,8 +134,10 @@ trait Erasure {
     def apply(tp: Type): Type = tp match {
       case FoldableConstantType(ct) =>
         // erase classOf[List[_]] to classOf[List]. special case for classOf[Unit], avoid erasing to classOf[BoxedUnit].
-        if (ct.tag == ClazzTag && ct.typeValue.typeSymbol != UnitClass) ConstantType(Constant(apply(ct.typeValue)))
-        else tp
+        if (ct.tag == ClazzTag)
+          if (ct.typeValue.typeSymbol == UnitClass) tp
+          else ConstantType(Constant(apply(ct.typeValue)))
+        else ct.tpe
```

Comment:
The fix for erasure. There are really 2 special cases under `Class` constant. Presumably the `Unit` case is handled specially later, but from this perspective, the encoding doesn't look regular.